File: Semester 2/APO C++/C++/Tests/V 1.2.0/_TUN/T_Tests_N.cpp
//
// IUT de Nice / Departement informatique / Module APO-C++
// Annee 2009_2010 - Charte des tests unitaires
//
// Exemple d'un module de tests unitaires limites a des cas nominaux
//
// Auteur : A. Thuaire
//
#include "..\_SRC\Tests.h"

void main () {
  
   Tests::Begin("Tests", "1.2.0");

      Tests::Design("Visualisation de niveau 0", 0);
         
         Tests::Case("C1 : 2 / OK"); {			
            Tests::Unit("x", "x");
            Tests::Unit(0, 0);
         }
      			       
      Tests::Design("Visualisation de niveau 1", 1);		
  
	     Tests::Case("C1 : 1 / OK"); {
            Tests::Unit("niveau 1", "niveau 1");
            Tests::Unit(1, 1);
         }

      Tests::Design("Visualisation de niveau 2", 2);		
  
         Tests::Case("C1 : 2 / OK"); {
            Tests::Unit("xxx", "xxx");
            Tests::Unit(2, 2);
         }

         Tests::Case("C2 : 3 / OK"); {
            Tests::Unit("niveau 1", "niveau 1");
            Tests::Unit(1, 1);
            Tests::Unit(-1,-1); 
         } 

      Tests::Design("Visualisation de niveau 3 avec TU tous OK", 3);		
  
	     Tests::Case("C1 : 3 / OK"); {
            Tests::Unit("......", "......");
            Tests::Unit(3, 3);
            Tests::Unit(3, 3);
         }

         Tests::Case("C2 : 2 / OK"); {
            Tests::Unit("xxxxx", "xxxxx");
            Tests::Unit(-3, -3);
         } 
            
         Tests::Case("C3 : 1 / OK"); {
            Tests::Unit(-3.0f, -3.0f);
         }  		
  
         Tests::Case("C1 : 2 / OK"); {
            Tests::Unit("......", "......");
            Tests::Unit(4, 4);
         }

         Tests::Case("C2 : 1 / OK !"); {
            Tests::Unit(-4.0f, -4.01f+0.01f);
         } 
            
         Tests::Case("C3 : 1 / OK"); {
            Tests::Unit(-4.25f, -4.25f);
         }

      Tests::Design("Modification du niveau verbose par un cas de test", 1);		
  
	     Tests::Case("C1 : 3 / OK"); {
            Tests::Unit("......", "......");
            Tests::Unit(3, 3);
            Tests::Unit(3, 3);
         }

         Tests::Case("C2 : 2 / OK", 2); {
            Tests::Unit("xxxxx", "xxxxx");
            Tests::Unit(-3, -3);
         } 
            
         Tests::Case("C3 : 1 / OK"); {
            Tests::Unit(-3.0f, -3.0f);
         }  		
  
         Tests::Case("C1 : 2 / OK", 3); {
            Tests::Unit("......", "......");
            Tests::Unit(4, 4);
         }

         Tests::Case("C2 : 1 / OK !"); {
            Tests::Unit(-4.0f, -4.01f+0.01f);
         } 
            
         Tests::Case("C3 : 1 / OK", 2); {
            Tests::Unit(-4.25f, -4.25f);
         }
             
   Tests::End();
}

File: Semester 2/APO_Java/Documents_techniques/Tests_unitaires/Tests/V 2.2.0/T_Tests_N.java
/**
*
* Annee 2009_2010 - Module Java - Charte des tests unitaires
*
* Classe Tests - Module de tests unitaires (cas nominaux)
*
* @author Alain Thuaire
*
**/
public class T_Tests_N {

   public static void main (String[] args) {
        
      Tests.Begin("Tests", "2.2.0");

      Tests.Design("Visualisation de niveau 0", 0);
         
         Tests.Case("C1 : 2 TU OK");			
            Tests.Unit("x", "x");
            Tests.Unit(new Integer(0), new Integer(0));
      			       
      Tests.Design("Visualisation de niveau 1", 1);		
  
         Tests.Case("C1 : 1 TU OK");
            Tests.Unit("niveau 1", "niveau 1");
            Tests.Unit(new Integer(1), new Integer(1));

      Tests.Design("Visualisation de niveau 2", 2);		
  
         Tests.Case("C1 : 2 TU OK");
            Tests.Unit("xxx", "xxx");
            Tests.Unit(new Integer(2), new Integer(2));

         Tests.Case("C2 : 3 TU OK");
            Tests.Unit("niveau 1", "niveau 1");
            Tests.Unit(new Integer(1), new Integer(1));
            Tests.Unit(new Integer(-1), new Integer(-1));  

      Tests.Design("Visualisation de niveau 3", 3);		
  
         Tests.Case("C1 : 3 TU OK");
            Tests.Unit("......", "......");
            Tests.Unit(new Integer(3), new Integer(3));
            Tests.Unit(new Float(3), new Float(3));

         Tests.Case("C2 : 2 TU OK");
            Tests.Unit("xxxxx", "xxxxx");
            Tests.Unit(new Integer(-3), new Integer(-3)); 
            
         Tests.Case("C3 : 1 TU OK");
            Tests.Unit(new Float(-3), new Float(-3));
            
      Tests.Design("Traitement des references nulles", 3);		
  
         Tests.Case("Deux references nulles");
            Tests.Unit(null, null);
                     
      Tests.End();

   }
}
